src/test/java/com/ververica/flink/table/gateway/operation/OperationWithUserJarTest.java

```diff
@@ -0,0 +1,137 @@
+import com.ververica.flink.table.gateway.utils.ResourceFileUtils;
+
+import org.apache.flink.client.cli.DefaultCLI;
+import org.apache.flink.client.deployment.DefaultClusterClientServiceLoader;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.table.api.DataTypes;
+import org.apache.flink.table.api.TableSchema;
+
+import org.junit.Test;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.Collections;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Test for operations when user jars are provided.
+ *
+ * <p>NOTE: before running this test, please make sure that {@code random-source-test-jar.jar}
+ * exists in the target directory. If not, run {@code mvn clean package} first.
+ */
+public class OperationWithUserJarTest extends OperationTestBase {
+
+	@Override
+	protected DefaultContext getDefaultContext() {
+		URL jarUrl = compileUserDefinedSource();
+		return new DefaultContext(
+			new Environment(),
+			Collections.singletonList(jarUrl),
+			new Configuration(),
+			new DefaultCLI(new Configuration()),
+			new DefaultClusterClientServiceLoader());
+	}
+
+	@Test
+	public void testView() {
+		createUserDefinedSource(context, "R");
+
+		CreateViewOperation createViewOperation = new CreateViewOperation(
+			context, "MyView1", "SELECT * FROM R");
+		ResultSet createViewResult = createViewOperation.execute();
+		assertEquals(OperationUtil.OK, createViewResult);
+
+		String[] tables1 = context.getExecutionContext().getTableEnvironment().listTables();
+		Arrays.sort(tables1);
+		assertArrayEquals(new String[]{"MyView1", "R"}, tables1);
+		assertTrue(context.getExecutionContext().getEnvironment().getTables().containsKey("MyView1"));
+
+		DropViewOperation operation2 = new DropViewOperation(context, "MyView1", false);
+		assertEquals(OperationUtil.OK, operation2.execute());
+
+		String[] tables2 = context.getExecutionContext().getTableEnvironment().listTables();
+		assertArrayEquals(new String[]{"R"}, tables2);
+	}
+
+	@Test
+	public void testDescribe() throws Exception {
+		createUserDefinedSource(context, "R");
+
+		DescribeOperation operation = new DescribeOperation(context, "R");
+		ResultSet resultSet = operation.execute();
+
+		TableSchema tableSchema = TableSchema.builder()
+			.field("a", DataTypes.INT())
+			.field("b", DataTypes.BIGINT())
+			.build();
+
+		DescribeOperationTest.compareResult(tableSchema, resultSet);
+	}
+
+	@Test
+	public void testExplain() {
+		createUserDefinedSource(context, "R");
+
+		ExplainOperation operation = new ExplainOperation(context, "select * from R");
+		ResultSet resultSet = operation.execute();
+
+		String expectedExplain = ResourceFileUtils.readAll(
+			"plan/operation-with-user-jar-test.test-explain.expected");
+		ExplainOperationTest.compareResult(expectedExplain, resultSet);
```

Comment:
It's better we do not depend on another Test class here, we can create an expected ResultSet instance and call assertEquals()